kratos/utilities/nurbs_utilities/projection_nurbs_geometry_utilities.h

```diff
@@ -0,0 +1,197 @@
+        *        point in case the projection is successful 
+        * @param rResult The projection onto the Nurbs curve geometry
+        * @param rNurbsCurve The Nurbs curve geometry onto which the point is 
+        *        to be projected
+        * @param MaxIterations Maximum number of iterations for the Newton-Rapshon 
+        *        algorithm
+        * @param ModelTolerance Tolerance of the CAD model
+        * @param Accuracy Accuracy for the the Newton-Rapshon algorithm
+        */
+        template <int TDimension, class TPointType>
+        bool NewtonRaphsonCurve(
+            CoordinatesArrayType& rParameter,
+            const CoordinatesArrayType& rPointGlobal,
+            CoordinatesArrayType& rResultLocal,
+            const NurbsCurveGeometry<TDimension, TPointType>& rNurbsCurve,
+            const int MaxIterations = 20,
+            const double Accuracy = 1e-6,
+            const double ModelTolerance = 1e-6)
+        {
+            // Intialize variables
+            double residual, delta_t;
+
+            // Loop over all Newton-Raphson iterations
+            for (int i = 0; i < MaxIterations; ++i) 
+            {
+                // Compute the position, the base and the acceleration vector
+                std::vector<array_1d<double, 3>> derivatives;
+                rNurbsCurve.GlobalSpaceDerivatives(
+                    derivatives,
+                    rParameter,
+                    2);
+                rResultLocal = derivatives[0];
+
+                // Compute the distance vector between the point and its 
+                // projection on the curve
+                array_1d<double, 3> distance_vector = rResultLocal - rPointGlobal;
+                if (norm_2(distance_vector) < std::max(ModelTolerance,Accuracy))
+                    return true;
+
+                // Compute the residual
+                residual = inner_prod(distance_vector, derivatives[1]);
+                if (std::abs(residual) < Accuracy)
+                    return true;
+
+                // Compute the increment
+                delta_t = residual / (inner_prod(derivatives[2], distance_vector) + pow(norm_2(derivatives[1]), 2));
+
+                // Increment the parametric coordinate
+                rParameter[0] -= delta_t;
+
+                // Check if the increment is too small and if yes return true
+                if (norm_2(delta_t*derivatives[1]) < Accuracy)
+                    return true;
+
+                // Check if the parameter gets out of its interval of definition and if so clamp it 
+                // back to the boundaries
+                rNurbsCurve.DomainInterval().IsInside(rParameter[0]);
+            }
+
+            // Return false if the Newton-Raphson iterations did not converge
+            return false;
+    }
+
+    template <int TDimension, class TPointType>
+    bool NewtonRaphsonSurface(
+        CoordinatesArrayType& rParameter,
```

Comment:
We could also name the Parameter to rParameterLocalCoordinates. I would accordingly name the other parameters LocalCoordinates and GlobalCoordinates. It's done like this also in the base class. Anyways, this is as you prefer. Could you put here also the comment from above?